src/server/lib/renderer.tsx

```diff
@@ -70,16 +80,24 @@ export const renderer: (url: string, opts: RendererOpts) => string = (
 
   const location = `${url}${queryString ? `?${queryString}` : ''}`;
 
+  const {
+    abTesting: { mvtId = 0, forcedTestVariants = {} } = {},
+  } = clientState;
+
   // Any changes made here must also be made to the hydration in the static webpack bundle
```

Comment:
Is it possible to abstract this definition so that is only needs to be changed in a single location ?

Reply:
it's just the nature of doing both SSR and CSR, we have to duplicate things in the 2 places so react is able to hydrate correctly after the initial server render.